src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimMatcher.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.civil.model.citizenui;
+
+import lombok.AllArgsConstructor;
+import uk.gov.hmcts.reform.civil.enums.CaseState;
+import uk.gov.hmcts.reform.civil.enums.RespondentResponsePartAdmissionPaymentTimeLRspec;
+import uk.gov.hmcts.reform.civil.enums.RespondentResponseTypeSpec;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+@AllArgsConstructor
+public class CcdDashboardClaimMatcher implements Claim {
+
+    private static final LocalTime FOUR_PM = LocalTime.of(16, 1, 0);
+    private CaseData caseData;
+
+    @Override
+    public boolean hasResponsePending() {
+        return caseData.getRespondent1ResponseDate() == null;
+    }
+
+    @Override
+    public boolean hasResponsePendingOverdue() {
+        return caseData.getRespondent1ResponseDeadline() != null && caseData.getRespondent1ResponseDeadline().isBefore(
+            LocalDate.now().atTime(FOUR_PM))
+            && caseData.hasBreathingSpace();
+    }
+
+    @Override
+    public boolean hasResponseDueToday() {
+        return caseData.getRespondent1ResponseDeadline() != null && caseData.getRespondent1ResponseDeadline().toLocalDate().isEqual(
+            LocalDate.now())
+            && caseData.getRespondent1ResponseDeadline().isBefore(LocalDate.now().atTime(FOUR_PM));
+    }
+
+    @Override
+    public boolean hasResponseFullAdmit() {
+        return caseData.getRespondent1ClaimResponseTypeForSpec() != null
+            && caseData.getRespondent1ClaimResponseTypeForSpec() == RespondentResponseTypeSpec.FULL_ADMISSION;
+    }
+
+    @Override
+    public boolean defendantRespondedWithFullAdmitAndPayImmediately() {
+        return hasResponseFullAdmit()
+            && RespondentResponsePartAdmissionPaymentTimeLRspec.IMMEDIATELY == caseData.getDefenceAdmitPartPaymentTimeRouteRequired();
```

Comment:
You're not checking for null here but are in other places - maybe this would be a better approach to use to rather than null checking everywhere else?

Reply:
all the enums that are to do with payment option i do like that. with others i think we still need to check whether the field is not null